book/writing-programs/cycle-tracking.md

```diff
@@ -56,69 +56,44 @@ fn main() {
+```sh
+    TRACE_FILE=output.json TRACE_SAMPLE_RATE=100 cargo run ...
 ```
-  [00:00:00] [########################################] 17053/17053 (0s)
-
-Total instructions in trace: 17053
-
-
- Instruction counts considering call graph
-+----------------------------------------+-------------------+
-| Function Name                          | Instruction Count |
-| __start                                | 17045             |
-| main                                   | 12492             |
-| sp1_zkvm::syscalls::halt::syscall_halt | 4445              |
-| sha2::sha256::compress256              | 4072              |
-| sp1_lib::io::commit                    | 258               |
-| sp1_lib::io::SyscallWriter::write      | 255               |
-| syscall_write                          | 195               |
-| memcpy                                 | 176               |
-| memset                                 | 109               |
-| sp1_lib::io::read_vec                  | 71                |
-| __rust_alloc                           | 29                |
-| sp1_zkvm::heap::SimpleAlloc::alloc     | 22                |
-| syscall_hint_len                       | 3                 |
-| syscall_hint_read                      | 2                 |
-+----------------------------------------+-------------------+
-
-
- Instruction counts ignoring call graph
-+----------------------------------------+-------------------+
-| Function Name                          | Instruction Count |
-| main                                   | 12075             |
-| sha2::sha256::compress256              | 4073              |
-| sp1_zkvm::syscalls::halt::syscall_halt | 219               |
-| memcpy                                 | 180               |
-| syscall_write                          | 123               |
-| memset                                 | 111               |
-| sp1_lib::io::commit                    | 88                |
-| sp1_lib::io::SyscallWriter::write      | 60                |
-| __start                                | 45                |
-| sp1_lib::io::read_vec                  | 35                |
-| sp1_zkvm::heap::SimpleAlloc::alloc     | 23                |
-| anonymous                              | 7                 |
-| __rust_alloc                           | 7                 |
-| syscall_hint_len                       | 4                 |
-| syscall_hint_read                      | 3                 |
-+----------------------------------------+-------------------+
+
+To view these profiles, we recommend [Samply](https://github.com/mstange/samply).
+```sh
+    cargo install --locked samply
+    samply load output.json
 ```
+
+Samply uses the Firefox profiler to create a nice visualization of your programs execution.
+![An example screenshot of the Firefox Profiler](../profiling.png)
+
+#### Interpreting the Profile
+- The "time" measurement in the profiler is actually the number of cycles spent, 
```

Comment:
this is good, it would be nice to explain what each view in the profiler is and how to interpret it, like call tree (invert call stack), flame graph, the timeline at the top

Reply:
agreed

Reply:
like some people might not know how to read the flame graph, for example

Reply:
or they might not know to invert call stack in order to find the most frequent functions